include/mim_control/impedance_controller.hpp

```diff
@@ -19,199 +19,243 @@ namespace mim_control
+
+        /** @brief (urdf) name of the root frame. The mim_control_ controller is
+         * computed with respect to this frame. */
+        std::string root_frame_name_;
+
+        /** @brief Index of the root frame in the pinocchio model. */
+        pinocchio::FrameIndex root_frame_index_;
+
+        /** @brief Measured root frame placement. */
+        pinocchio::SE3 root_placement_;
+
+        /** @brief Measured root frame orientation. */
+        pinocchio::SE3 root_orientation_;
+
+        /** @brief Measured root frame velocity. */
+        pinocchio::Motion root_velocity_;
+
+        /** @brief (urdf) name of the end frame. This is the controlled frame. */
+        std::string end_frame_name_;
+
+        /** @brief Index of the end frame in the pinocchio model. */
+        pinocchio::FrameIndex end_frame_index_;
+
+        /** @brief Jacobian of the end frame. */
+        pinocchio::Data::Matrix6x end_jacobian_;
+
+        /** @brief Measured end frame placement. */
+        pinocchio::SE3 end_placement_;
+
+        /** @brief Measured end frame orientation. */
+        pinocchio::SE3 end_orientation_;
+
+        /** @brief Measured end frame velocity. */
+        pinocchio::Motion end_velocity_;
+
+        /** @brief Impedance force. Accessible for machine learning purposes. */
+        Vector6d impedance_force_;
+
+        /** @brief Measured end frame placement in root frame. */
+        pinocchio::SE3 actual_end_frame_placement_;
+
+        /** @brief Measured end frame placement in root frame. */
+        pinocchio::Motion actual_end_frame_velocity_;
+
+        /** @brief SE3 placement error. */
+        Eigen::Matrix<double, 6, 1> err_se3_;
+
+        /** @brief SE3 velocity error. */
+        pinocchio::Motion err_vel_;
+
+        /** @brief Jacobian used in the computation of the mim_control_. */
+        pinocchio::Data::Matrix6x impedance_jacobian_;
+
+        /** @brief Output torques. */
+        Eigen::VectorXd torques_;
+
+        /** @brief Output torques. */
+        Eigen::VectorXd joint_torques_;
+
+        /** @brief Checks out if the Pinocchio rigid body model of the robot
+         * contains a free-flyer. This is used to return the command i.e. the
+         * torques to be applied to the joints only. */
+        bool pinocchio_model_has_free_flyer_;
+
+        /** @brief Stores the root jacobian */
+        pinocchio::Data::Matrix6x root_jacobian_;
 
-    /** @brief Checks out if the Pinocchio rigid body model of the robot
-     * contains a free-flyer. This is used to return the command i.e. the
-     * torques to be applied to the joints only. */
-    bool pinocchio_model_has_free_flyer_;
-};
+        /** @brief Relative velocity between the end_frame with respect to a frame
+         * whose origin aligns with the root frame but is oriented as the world frame. */
+        Eigen::VectorXd relative_vel_between_frames;
+    };
 
-}  // namespace mim_control
+}  // namespace mim_control_
```

Comment:
Any reason for changing this line :thinking:

Reply:
No reason, I think I did a search and replace which was wrong. I reverted it back.